inductor: fix onednn conv2d(transpose) packed issue when input size is three

XiaobingSuper opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99696
* __->__ #99601

Fix https://github.com/pytorch/pytorch/issues/99568, this PR adds an input size check before doing conv2d(transpose) packed.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

XiaobingSuper committed:
inductor: fix onednn conv2d(transpose) packed issue when input size is three

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor: fix onednn conv2d(transpose) packed issue when input size is three"


Fix https://github.com/pytorch/pytorch/issues/99568, this PR adds an input size check before doing conv2d(transpose) packed.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jgong5 commented on torch/_inductor/mkldnn.py:
```diff
@@ -295,9 +295,9 @@ def pack_module(gm: torch.fx.GraphModule):
                     computation_node_input_size = None
-                    if (
+                    if not (
                         type(cur_module) in [torch.nn.Linear]
-                        and cur_module.weight.dtype == torch.float32
+                        and cur_module.weight.dtype == torch.bfloat16
```
Suggest to add comment on why we only enable prepacking for bfloat16 linear on dynamic shapes.

XiaobingSuper replied:
added

XiaobingSuper committed:
Update on "inductor: fix onednn conv2d(transpose) packed issue when input size is three"


Fix https://github.com/pytorch/pytorch/issues/99568, this PR adds an input size check before doing conv2d(transpose) packed.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper commented:
@pytorchbot merge
